Assignment2/Q1/src/BinaryTree.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Created by Alice Colt.
+ *
+ * Implementaiton of a Binary Tree with functions for insert, serach and takeAncestors
+ *
+ */
+public class BinaryTree {
+    private NodeBinaryTree root;
+
+    //Constructor
+    public BinaryTree(NodeBinaryTree new_root){
+        root = new_root;
+    }
+
+
+    public void insert(NodeBinaryTree new_node) {
+
+        NodeBinaryTree actualNode = root;
+
+        if (new_node != null) {
+
+            while (actualNode != null) {
+                if (new_node.compareTo(actualNode) == 1 && actualNode.getLeft() == null) {
```

Comment:
Note that the compareTo interface doesn't guarantee that you get -1, 0, or +1 - it could be any number. I know that your implementation returns -1, 0 or +1, but more defensive programming would use < 0, == 0, > 0.